onnxruntime/core/framework/session_state.cc

```diff
@@ -1098,12 +1098,52 @@ Status SessionState::FinalizeSessionState(const std::basic_string<PATH_CHAR_TYPE
                                   remove_initializers, constant_initializers_use_count);
 }
 
+static OrtValueIndex Index(const OrtValueNameIdxMap& ort_value_name_idx_map,
+                           const OrtValueName& name) {
+  OrtValueIndex result;
+  auto status = ort_value_name_idx_map.GetIdx(name, result);
+  ORT_ENFORCE(status.IsOK(), status.ErrorMessage());
+  return result;
+}
+
+static Status OuterScopeNodeArgLocationAccumulator(const SequentialExecutionPlan& plan,
+                                                   const OrtValueNameIdxMap& ort_value_name_to_idx_map,
```

Comment:
Please add comments describing what this does and why

Reply:
done